thehive/app/org/thp/thehive/services/TaskSrv.scala

```diff
@@ -114,6 +114,36 @@ class TaskSrv @Inject() (caseSrvProvider: Provider[CaseSrv], auditSrv: AuditSrv)
       _ <- auditSrv.task.update(task, Json.obj("assignee" -> user.login))
     } yield ()
   }
+
+  def isActionRequired(
+    task: Task with Entity,
+    organisations: Seq[Organisation with Entity]
+  )(implicit graph: Graph): Map[String, Boolean] = {
+    organisations
```

Comment:
This function executes as many queries as number of organisations (from parameter). It could be more efficient like that:
```suggestion
    get(task)
      .inE[ShareTask]
      .project(
        _.by(_.outV.v[Share].organisation.hasId(organisations.map(_._id): _*).value(_.name))
          .byValue(_.actionRequired)
      )
      .toSeq
      .toMap
```